scalyr_agent/scalyr_monitor.py

```diff
@@ -120,6 +121,9 @@ def __init__(
         # The logger instance that this monitor should use to report all information and metric values.
         self._logger = logger
         self.monitor_name = monitor_config["module"]
+        self.short_hash = hashlib.sha256(self.monitor_name.encode("utf-8")).hexdigest()[
```

Comment:
IIRC, the `monitor_name` is guaranteed to be unique because we get the instance-specific name suffix from its logger, right?

Reply:
IIRC, ``monitor_name`` here contains just the fully qualified name to the monitor class and the value which is used for logging and contains the id as well is different, but I'll dig in.

Reply:
Yep, it appears we only pass ``id`` to the logger instance (https://github.com/scalyr/scalyr-agent-2/blob/master/scalyr_agent/monitors_manager.py#L508) and we don't have access to it in the monitor class.

Having said that, I could also refactor the code to pass the ``id`` to the monitor class and use that instead of the hash.

EDIT: This actually brings up good point - we want to use ``<monitor module>.<monitor instance id>.<metric name>`` for the internal dict key and not just ``<monitor module>.<metric name>``.

So it turns out we actually need both :)